packages/wallet-sdk/src/util/provider.ts

```diff
@@ -43,6 +43,16 @@ export function getCoinbaseInjectedSigner(): Signer | undefined {
   return window.coinbaseWalletSigner;
 }
 
+function getInjectedEthereum(): CBInjectedProvider | undefined {
+  const window = globalThis as CBWindow;
+  try {
+    return window.ethereum ?? window.top?.ethereum;
+    // eslint-disable-next-line no-empty
+  } catch {}
+
+  return undefined;
+}
+
 export function getCoinbaseInjectedProvider({
   metadata,
   preference,
```

Comment:
we might remove the line 50, 
and do this instead for line 56, `const extension =(globalThis as CBWindow).coinbaseWalletExtension;`

Reply:
or, seems bit overkill, but for the sake of consistency, we might create a dedicated helper method for this maybe?